doctr/io/pdf.py

```diff
@@ -42,4 +42,5 @@ def read_pdf(
 
     # Rasterise pages to PIL images with pypdfium2 and convert to numpy ndarrays
     with pdfium.PdfDocument(file, password=password) as pdf:
-        return [np.asarray(img) for img in pdf.render_topil(scale=scale, **kwargs)]
+        for img in pdf.render_topil(scale=scale, **kwargs):
+            yield np.asarray(img)
```

Comment:
May be a one-liner:
```python3
yield from pdf.render_topil(...)
```

Reply:
Mh looks a bit strange 😅 
`yield from np.asarray(pdf.render_topil(...))`

Reply:
Right, I see! Overlooked that in the hurry. I expect it'll be better once I have released v3 so that you can use `render_tonumpy()`, see https://github.com/mindee/doctr/discussions/1032.

Reply:
A got it would you link the discussion to the issue ? :) So i would close this PR until your release is finally